packages/pigeon/mock_handler_tester/test/widget_test.dart

```diff
@@ -73,28 +73,23 @@ void main() {
       final Mock mock = Mock();
       TestHostApi.setup(mock);
       expect(
-        await const BasicMessageChannel<Object>(
+        await const BasicMessageChannel<Object?>(
           'dev.flutter.pigeon.Api.initialize',
           StandardMessageCodec(),
         ).send(null),
         isEmpty,
       );
-      try {
-        await const BasicMessageChannel<Object>(
-          'dev.flutter.pigeon.Api.search',
-          StandardMessageCodec(),
-        ).send(null) as Map<Object, Object>;
-        expect(true, isFalse); // should not reach here
-      } catch (error) {
-        expect(error, isAssertionError);
-        expect(
-          error.toString(),
-          contains(
-            'Argument for dev.flutter.pigeon.Api.search was null. Expected SearchRequest.',
-          ),
-        );
-      }
-      expect(mock.log, <String>['initialize']);
+      final Object? result = await const BasicMessageChannel<Object?>(
+        'dev.flutter.pigeon.Api.search',
+        StandardMessageCodec(),
+      ).send(null);
+      expect(result, isA<Map<Object?, Object?>>());
+      final Map<Object?, Object?>? resultMap = result as Map<Object?, Object?>?;
+      expect(
+          (resultMap?['result'] as Map<Object?, Object?>?)?['result'], isNull);
+      expect(
+          (resultMap?['result'] as Map<Object?, Object?>?)?['error'], isNull);
+      expect(mock.log, <String>['initialize', 'search']);
```

Comment:
Ahh, I see you changed the behavior too.  This is a breaking change and we should avoid it.  (Although I know there is a weird issue with this code on the stable branch, on master it works fine).

Reply:
reverted as noted in above comment